Add vim mode spike

Kethku opened the pull request:
Also refactors editor movement actions to use common utility functions

Commit:
Add vim mode spike
Refactor editor movement actions to use common utility functions
Add global update notifications

Co-authored-by: Nathan Sobo <nathan@zed.dev>
Co-authored-by: Max Brunsfeld <max@zed.dev>

maxbrunsfeld reviewed (commented):
Seems like you have quite a lot of the main commands implemented! Excited to try this out.

nathansobo committed:
Replace move_and_delete with transact

It's more general and reusable at the cost of a bit more duplication in these other cases. I think it's reasonable to perform these steps separately always.
